pkg/network/server.go

```diff
@@ -423,6 +426,10 @@ func (s *Server) run() {
 		addrTimer        = time.NewTimer(peerCheckTime)
 		peerTimer        = time.NewTimer(s.ProtoTickInterval)
 	)
+	s.runLoopStarted.Store(true)
```

Comment:
BTW, you can still have arbitrarily long delays between this and any of the `for` logic. You can however add some channel to send something from `Start` into `run` main `select` (and forget this channel after the first event from it).

Reply:
> add some channel to send something from Start into run main select

This won't work since `run` is not a separate routine, we can't send from `Start` and call `run` at the same time. May be done if we send `run` to a separate routine from `Start`, but `Start` is already a separate routine from POW of other services, thus we'll need to refactor all callers of `Start`. Agree to implement?

Reply:
Agreed, implemented.